cognite/client/_api/spaces.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+from typing import Iterator, Optional, Sequence, cast, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import LIST_LIMIT_DEFAULT
+from cognite.client.data_classes import Space, SpaceList
+from cognite.client.utils._identifier import DataModelingIdentifierSequence
+
+
+class SpacesAPI(APIClient):
+    _RESOURCE_PATH = "/models/spaces"
+
+    def __call__(
+        self,
+        chunk_size: int = None,
+        limit: int = None,
+    ) -> Iterator[Space] | Iterator[SpaceList]:
+        """Iterate over spaces
+
+        Fetches spaces as they are iterated over, so you keep a limited number of spaces in memory.
+
+        Args:
+            chunk_size (int, optional): Number of spaces to return in each chunk. Defaults to yielding one space a time.
+            limit (int, optional): Maximum number of spaces to return. Default to return all items.
+
+        Yields:
+            Union[Space, SpaceList]: yields Space one by one if chunk_size is not specified, else SpaceList objects.
+        """
+        return self._list_generator(
+            list_cls=SpaceList,
+            resource_cls=Space,
+            method="GET",
+            chunk_size=chunk_size,
+            limit=limit,
+        )
+
+    def __iter__(self) -> Iterator[Space]:
+        """Iterate over spaces
+
+        Fetches spaces as they are iterated over, so you keep a limited number of spaces in memory.
+
+        Yields:
+            Space: yields Spaces one by one.
+        """
+        return cast(Iterator[Space], self())
+
+    def retrieve(self, space: str) -> Optional[Space]:
+        """`Retrieve a single space by id. <https://docs.cognite.com/api/v1/#tag/Spaces/operation/bySpaceIdsSpaces>`_
+
+        Args:
+            space (str): Space ID
+
+        Returns:
+            Optional[Space]: Requested space or None if it does not exist.
+
+        Examples:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.models.spaces.retrieve(space='mySpace')
```

Comment:
There are ~30 occurrences that needs renaming:
```suggestion
                >>> res = c.data_modeling.spaces.retrieve(space='mySpace')
```